quail/processes/wps_climdex_gsl.py

```diff
@@ -0,0 +1,135 @@
+import os
+from rpy2 import robjects
+from pywps import Process, LiteralInput
+from pywps.app.Common import Metadata
+
+from wps_tools.utils import log_handler, collect_args, common_status_percentages
+from wps_tools.io import log_level
+from quail.utils import get_package, logger, load_rdata_to_python, save_python_to_rdata
+from quail.io import climdex_input, ci_name, output_file, rda_output, vector_name
+
+
+class ClimdexGSL(Process):
+    """
+    Computes the growing season length (GSL): Growing season length
+    is the number of days between the start of the first spell of warm days
+    in the first half of the year, defined as six or more days with mean
+    temperature above 5 degrees Celsius, and the start of the first spell
+    of cold days in the second half of the year, defined as six or more days
+    with a mean temperature below 5 degrees Celsius
+    """
+
+    def __init__(self):
+        self.status_percentage_steps = dict(
+            common_status_percentages,
+            **{
+                "load_rdata": 10,
+                "save_rdata": 90,
+            },
+        )
+        inputs = [
+            climdex_input,
+            ci_name,
+            output_file,
+            vector_name,
+            LiteralInput(
+                "gsl_mode",
+                "GSL mode",
+                abstract="Growing season length method to use. The three alternate modes provided ('GSL_first', 'GSL_max', and 'GSL_sum') are for testing purposes only.",
```

Comment:
Why exactly do we have these testing modes?

Reply:
It was just an available parameter, I wasn't 100% sure whether I should add it.
```quote
’GSL_first’ returns the first growing sea-son found; ’GSL_max’ returns the 
longest growing season found; and ’GSL_sum’ returns the total length of all
growing seasons found
```